### PR TITLE
pynitrokey: init at 0.4.9

### PR DESCRIPTION
**Pull request by frogamic**
###### Motivation for this change

pynitrokey is the CLI app for managing nitrokey devices

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by devhell** [comment]
This is very cool, thank you!

File: pkgs/tools/security/pynitrokey/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  version = "0.4.9";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-mhH6mVgLRX87PSGTFkj1TE75jU1lwcaRZWbC67T+vWo=";
+  };
+
+  doCheck = false;
+
+  format = "flit";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pynitrokey/default.nix:35]:
```suggestion
```

File: pkgs/tools/security/pynitrokey/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      tqdm
+      intelhex
+      libusb1
+    ];
+  };
+in
+
+buildPythonApplication rec {
+  pname = "pynitrokey";
+  version = "0.4.9";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pynitrokey/default.nix:26]:
```suggestion
  version = "0.4.9";
  format = "flit";
```

File: pkgs/tools/security/pynitrokey/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    pyserial
+    pyusb
+    requests
+    pygments
+    python-dateutil
+    urllib3
+    cffi
+    cbor
+    nkdfu
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pynitrokey/default.nix:51]:
```suggestion
  ];

  # no tests
  doCheck = false;

  pythonImportsCheck = [ "pynitrokey" ];
```

File: pkgs/tools/security/pynitrokey/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    };
+
+    format = "flit";
+    propagatedBuildInputs = [
+      fire
+      tqdm
+      intelhex
+      libusb1
+    ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pynitrokey/default.nix:21]: Can we move this to pythonPackages?

Comment (frogamic) [reply pkgs/tools/security/pynitrokey/default.nix:21]: Can I add 2 packages in 1 commit? or should create 2 `init at x.y.z` commits?

Comment (SuperSandro2000) [reply pkgs/tools/security/pynitrokey/default.nix]: Please do two commits. Can be all done in this PR.

File: pkgs/development/python-modules/nkdfu/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ python3Packages, lib }:
+
+with python3Packages;
+
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nkdfu/default.nix:5]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, fire, tqdm, intelhex, libusb1 }:

buildPythonPackage rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/nkdfu/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    inherit pname version;
+    sha256 = "sha256-Y8GonfCBi3BNMhZ99SN6/SDSa0+dbfPIMPoVzALwH5A=";
+  };
+
+  propagatedBuildInputs = [
+    fire
+    tqdm
+    intelhex
+    libusb1
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nkdfu/default.nix:18]: The package has no tests. We should add a `doCheck = false` and `pythonImportsCheck = [ "nkdfu" ];`.

**Pull request merged**